pennylane_qiskit/converter.py

```diff
@@ -256,3 +296,120 @@ def load_qasm_from_file(file: str):
         function: the new PennyLane template
     """
     return load(QuantumCircuit.from_qasm_file(file))
+
+
+# diagonalize is currently only used if measuring
+# maybe always diagonalize when measuring, and never when not?
+# will this be used for a user-facing function to convert from PL to Qiskit as well?
+def circuit_to_qiskit(circuit, register_size, diagonalize=True, measure=True):
+    """Builds the circuit objects based on the operations and measurements
+    specified to apply.
+
+    Args:
+        circuit (Union[~.QuantumScript, ~.QuantumTape]): the circuit applied
+            to the device
+        register_size (int): the total number of qubits on the device the circuit is
+            executed on; this must include any qubits not used in the given
+            circuit to ensure correct indexing of the returned samples
+
+    Keyword args:
+        diagonalize (bool): whether or not to apply diagonalizing gates before the
+            measurements
+        measure (bool): whether or not to apply measurements at the end of the circuit;
+            a full circuit is represented either as a Qiskit circuit with operations
+            and measurements (measure=True), or a Qiskit circuit with only operations,
+            paired with a Qiskit Estimator defining the measurement process.
+    """
+
+    reg = QuantumRegister(register_size)
+
+    if not measure:
+        qc = QuantumCircuit(reg, name="temp")
+
+        for op in circuit.operations:
+            qc &= operation_to_qiskit(op, reg)
+
+        return qc
+
+    creg = ClassicalRegister(register_size)
+    qc = QuantumCircuit(reg, creg, name="temp")
+
+    for op in circuit.operations:
+        qc &= operation_to_qiskit(op, reg, creg)
+
+    # rotate the state for measurement in the computational basis
+    if diagonalize:
```

Comment:
follow up: check this in cases with multiple different bases

Reply:
Will address in a separate PR in the future